pkg/v1/tkr/controllers/source/tkr_source_controller.go

```diff
@@ -347,9 +347,7 @@ func (r *reconciler) UpdateTKRCompatibleCondition(ctx context.Context, tkrs []ru
 
 	compatibleReleases := []string{}
 	for _, mgmtVersion := range metadata.ManagementClusterVersions {
-		// Fix before TKG v1.10: what if mgmtClusterVersion is "v1.10" and mgmtVersion.TKGVersion is "v1.1"?
-		// See https://github.com/vmware-tanzu/tanzu-framework/issues/452
-		if strings.HasPrefix(mgmtClusterVersion, mgmtVersion.TKGVersion) {
+		if mgmtClusterVersion == mgmtVersion.TKGVersion {
```

Comment:
Do we have any sense of why this was a prefix match previously, and why this wasn't just moved to exact match instead of adding the TODO?

Reply:
I _think_ this was initial limitation from release team where the management cluster version(fetched from tkg BOM) for dev builds was superset of the version published in tkr-compatibility. But later the release team had removed this limitation.
cc: @ZihanJiang96